File: 1221. Split a String in Balanced Strings/Solution.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        
        l_cnt, r_cnt = 0, 0
        ret = 0
        
        for ch in s:
            if ch == 'L':
                l_cnt += 1
            if ch == 'R':
                r_cnt += 1
            if l_cnt == r_cnt:
                ret += 1
                l_cnt, r_cnt = 0, 0
        
        return ret
        

File: 1221. Split a String in Balanced Strings/Solution.cpp
class Solution {
public:
    int balancedStringSplit(string s) {
        
        int l_cnt = 0;
        int r_cnt = 0;
        int ret = 0;
        
        for (auto &ch : s)
        {
            if (ch == 'L')
            {
                l_cnt++;
            }
            
            if (ch == 'R')
            {
                r_cnt++;
            }
            
            if (l_cnt == r_cnt)
            {
                l_cnt = 0;
                r_cnt = 0;
                ret ++;                    
            }
        
        return ret;
    }
};
